pkg/storage/secret/encryption/encrypted_value_store_test.go

```diff
@@ -198,6 +242,135 @@ func TestEncryptedValueStoreImpl(t *testing.T) {
 	})
 }
 
+func TestEncryptedValueMigration(t *testing.T) {
```

Comment:
I think in this case it could make sense to test edge cases of the migrator with a mock of the `ListAll` returning new data, maybe broken data, to exercise those code paths to see if they're matching what we expect.